spec/integration/chunker_spec.rb

```diff
@@ -169,6 +171,8 @@ def log_messages
     it 'should copy 23 rows from origin to destination in one shot with slave lag based throttler, regardless of the value of the id' do
       23.times { |n| execute("insert into origin set id = '#{ 100000 + n * n + 23 }'") }
 
+      ActiveRecord::Base.stubs(:connection_pool).returns(stub(spec: stub(config: @active_record_config)))
```

Comment:
Wait if you didn't find a root cause how did you solve the problem? Is this simply a workaround? Preferably we should understand what we are doing and what side effects it may have before writing the code to workaround the problem.

Reply:
I investigated and found that the error was being received from a MySQL2 client being used earlier in development. However, in a later change the main connection was changed to ActiveRecord, which doesn't need to stub `#connection_pool`.